storagemarket/types.go

```diff
@@ -59,3 +75,126 @@ type StorageProvider interface {
+	GetBalance(ctx context.Context, addr address.Address) (Balance, error)
+
+	// Publishes deal on chain
+	PublishDeals(ctx context.Context, deal MinerDeal) (DealID, cid.Cid, error)
+
+	// ListProviderDeals lists all deals associated with a storage provider
+	ListProviderDeals(ctx context.Context, addr address.Address) ([]StorageDeal, error)
+
+	// Called when a deal is complete and on chain, and data has been transferred and is ready to be added to a sector
+	// returns sector id
+	OnDealComplete(ctx context.Context, deal MinerDeal, piecePath string) (uint64, error)
+
+	// returns the worker address associated with a miner
+	GetMinerWorker(ctx context.Context, miner address.Address) (address.Address, error)
+
+	// Signs bytes
+	SignBytes(ctx context.Context, signer address.Address, b []byte) (*types.Signature, error)
+}
+
+type DealSectorCommittedCallback func(error)
+
+// Node dependencies for a StorageClient
+type StorageClientNode interface {
+	MostRecentStateId(ctx context.Context) (StateKey, error)
+
+	// Adds funds with the StorageMinerActor for a storage participant.  Used by both providers and clients.
+	AddFunds(ctx context.Context, addr address.Address, amount BigInt) error
+
+	EnsureFunds(ctx context.Context, addr address.Address, amt types.BigInt) error
+
+	// GetBalance returns locked/unlocked for a storage participant.  Used by both providers and clients.
+	GetBalance(ctx context.Context, addr address.Address) (Balance, error)
+
+	//// ListClientDeals lists all on-chain deals associated with a storage client
+	ListClientDeals(ctx context.Context, addr address.Address) ([]StorageDeal, error)
+
+	// GetProviderInfo returns information about a single storage provider
+	//GetProviderInfo(stateId StateID, addr Address) *StorageProviderInfo
+
+	// GetStorageProviders returns information about known miners
+	ListStorageProviders(ctx context.Context) ([]*StorageProviderInfo, error)
+
+	// Subscribes to storage market actor state changes for a given address.
+	// TODO: Should there be a timeout option for this?  In the case that we are waiting for funds to be deposited and it never happens?
+	//SubscribeStorageMarketEvents(addr Address, handler StorageMarketEventHandler) (SubID, error)
+
+	// Cancels a subscription
+	//UnsubscribeStorageMarketEvents(subId SubID)
+	ValidatePublishedDeal(ctx context.Context, deal ClientDeal) (uint64, error)
+
+	// SignProposal signs a proposal
+	SignProposal(ctx context.Context, signer address.Address, proposal *actors.StorageDealProposal) error
+
+	GetDefaultWalletAddress(ctx context.Context) (address.Address, error)
+
+	OnDealSectorCommitted(ctx context.Context, provider address.Address, dealId uint64, cb DealSectorCommittedCallback) error
+
+	ValidateAskSignature(ask *StorageAsk) error
+}
+
+type StorageClientProofs interface {
+	//GeneratePieceCommitment(piece io.Reader, pieceSize uint64) (CommP, error)
+}
+
+// Closely follows the MinerInfo struct in the spec
```

Comment:
For the future: we should figure out if we're gonna track MinerInfo cause it's changed a lot already in spec, or if this is a seperate data structure. I called it something separate when I wrote the storage market spec.